dali/pipeline/data/tensor_list.h

```diff
@@ -154,8 +164,14 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
       this->meta_[i].SetSkipSample(other[i].ShouldSkipSample());
     }
 
+    // Copy kernel only makes sense when copying to/from device or host pinned memory
     use_copy_kernel &= (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned()) &&
                        (std::is_same<Backend, GPUBackend>::value || pinned_);
+
+    // Copy kernel is the default behavior when copying from non-contiguous D2D
+    use_copy_kernel |= (std::is_same<SrcBackend, GPUBackend>::value && !other.IsContiguous()) &&
```

Comment:
Do we need this as it is already up to the user choice inside `ShareUserData`?
I would not touch tensor_list.h  here and leave that logic to the caller of the `Copy` fucntion.

Reply:
I reverted it. Now the flag from the  user is passed directly to ShareUserData